File: 卡码/代码随想录/Go/二叉树/02-二叉树属性/LeetCode_0112.go
//112. 路径总和
package main

import (
	. "binaryTree/types"
)

func hasPathSum(root *TreeNode, targetSum int) bool {
	if root==nil {
		return false
	}
	//如果当前节点是叶子节点且目标和与叶子节点值相等则找到路径
	if root.Left==nil&&root.Right==nil&&targetSum==root.Val {
		return true
	}
	return hasPathSum(root.Left,targetSum-root.Val)||hasPathSum(root.Right,targetSum-root.Val)
}

File: 卡码/代码随想录/C++/二叉树/02-二叉树属性/LeetCode_0112.cpp
//112. 路径总和
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  bool hasPathSum(TreeNode* root, int targetSum) {
		if(!root) return false;
		//如果当前节点是叶子节点且目标和与叶子节点值相等则找到路径
		if(!root->left&&!root->right&&targetSum==root->val) return true;
		return hasPathSum(root->left,targetSum-root->val)||hasPathSum(root->right,targetSum-root->val);
  }
};

int main(){
  system("pause");
  return 0;
}
